tm2py/emme/matrix.py

```diff
@@ -262,3 +262,91 @@ def read_hdf5(self, path: str) -> NumpyArray:
             Numpy array from OMX file
         """
         return self._omx_file.get_node(path).read()
+
+
+class TollChoiceCalculator:
+    """Implements toll choice calculations.
+
+    Centralized implementation of Toll Choice calculations common to
+    Commercial and Internal-external sub models. Loads input skims
+    from OMXManager
+
+    Args:
+        value_of_time: value of time to use in the utility expression
+        coeff_time: coefficient of time value used in the utility expression
+        operating_cost_per_mile: operating cost value in cents per mile
+            for converting distance to cost
+    """
+
+    def __init__(
```

Comment:
docstrings for __init__ args go here. 
docstrings for instance attributes go in class-level docstrings